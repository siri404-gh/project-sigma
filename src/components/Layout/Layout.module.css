.contentWrapper {
  bottom: 0;
  display: flex;
  flex-direction: column;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  z-index: -1;
}

.shift {
  max-height: 64px;
  min-height: 64px;
}
