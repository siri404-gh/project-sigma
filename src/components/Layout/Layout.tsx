import React, { useState, FC } from 'react'

import { Container } from '@mui/material'

import Navbar, { NavbarProps } from '@/components/Navbar/Navbar'
import Navlinks, { NavlinksProps } from '@/components/Navlinks/Navlinks'
import Seo, { SEOProps } from '@/components/Seo/Seo'
import Sidebar, { SidebarProps } from '@/components/Sidebar/Sidebar'

export interface LayoutProps {
  children?: JSX.Element
  seoProps?: SEOProps
  navbarProps?: NavbarProps
  navlinksProps?: NavlinksProps
  sidebarProps?: SidebarProps
}

const Layout: FC<LayoutProps> = ({
  children,
  seoProps,
  navbarProps,
  navlinksProps,
  sidebarProps,
}) => {
  const [isSidebarOpen, setIsSidebarOpen] = useState(false)
  const onSidebarToggle = () => setIsSidebarOpen(!isSidebarOpen)

  navbarProps = {
    ...navbarProps,
    onMenuClick: onSidebarToggle,
  }
  sidebarProps = {
    ...sidebarProps,
    isOpen: isSidebarOpen,
    onClose: onSidebarToggle,
    onOpen: onSidebarToggle,
  }

  return (
    <div className='layout'>
      <Seo {...seoProps} />
      <Sidebar {...sidebarProps} />
      <Container maxWidth='md' sx={{ height: '100vh' }} disableGutters>
        <Navbar {...navbarProps}>
          <Navlinks {...navlinksProps} />
        </Navbar>
        {children}
      </Container>
    </div>
  )
}

export default Layout
