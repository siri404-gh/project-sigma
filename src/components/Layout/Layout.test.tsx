import * as React from 'react'

import { render } from '@testing-library/react'

import Layout from '@/components/Layout/Layout'

xdescribe('Layout', () => {
  test('renders', () => {
    const { container } = render(<Layout />)
    expect(container).toMatchSnapshot()
  })
})
