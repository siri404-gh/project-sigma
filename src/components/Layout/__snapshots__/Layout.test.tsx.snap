// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders 1`] = `
<div>
  <div
    class="layout"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd MuiContainer-disableGutters css-1l7cv8f-MuiContainer-root"
    >
      <nav
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-pivuyh-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular toolbar css-1t29gy6-MuiToolbar-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1oaaite-MuiTypography-root-MuiLink-root"
            href="/"
          >
            <h1
              class="MuiTypography-root MuiTypography-h6 css-oblp1g-MuiTypography-root"
            />
          </a>
          <ul
            class="MuiBox-root css-4s7zdz"
          />
          <div
            class="MuiBox-root css-1yrwlx0"
          >
            <button
              aria-controls="simple-menu"
              aria-hidden="true"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AccountCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="menu"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                data-testid="MenuIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </nav>
    </div>
  </div>
</div>
`;
